includes/Checker/Checks/Performance/Non_Blocking_Scripts_Check.php

```diff
@@ -0,0 +1,241 @@
+	 * @throws Exception Thrown when a post type URL cannot be retrieved.
+	 */
+	protected function get_urls() {
+		$urls = array( home_url() );
+
+		foreach ( $this->get_viewable_post_types() as $post_type ) {
+			$posts = get_posts(
+				array(
+					'posts_per_page' => 1,
+					'post_type'      => $post_type,
+					'post_status'    => array( 'publish', 'inherit' ),
+				)
+			);
+
+			if ( ! isset( $posts[0] ) ) {
+				throw new Exception(
+					sprintf(
+						/* translators: %s: The Post Type name. */
+						__( 'Unable to retrieve post URL for post type: %s', 'plugin-check' ),
+						$post_type
+					)
+				);
+			}
+
+			$urls[] = get_permalink( $posts[0] );
+		}
+
+		return $urls;
+	}
+
+	/**
+	 * Amends the given result by running the check for the given URL.
+	 *
+	 * @since 1.1.0
+	 *
+	 * @param Check_Result $result The check result to amend, including the plugin context to check.
+	 * @param string       $url    URL to run the check for.
+	 *
+	 * @throws Exception Thrown when the check fails with a critical error (unrelated to any errors detected as part of
+	 *                   the check).
+	 *
+	 * @SuppressWarnings(PHPMD.NPathComplexity)
+	 */
+	protected function check_url( Check_Result $result, $url ) {
+		// Reset the WP_Scripts instance.
+		unset( $GLOBALS['wp_scripts'] );
+
+		// Run the 'wp_enqueue_script' action, wrapped in an output buffer in case of any callbacks printing scripts
+		// directly. This is discouraged, but some plugins or themes are still doing it.
+		ob_start();
+		wp_enqueue_scripts();
+		wp_scripts()->do_head_items();
+		wp_scripts()->do_footer_items();
+		ob_end_clean();
+
+		foreach ( wp_scripts()->done as $handle ) {
+			$script = wp_scripts()->registered[ $handle ];
+
+			// TODO: Somehow detect inline scripts added by the plugin that don't have a `src`.
+
+			if ( ! $script->src || strpos( $script->src, $result->plugin()->url() ) !== 0 ) {
+				continue;
+			}
+
+			if ( ! empty( $script->extra['strategy'] ) ) {
+				continue;
+			}
+
+			$script_path = str_replace( $result->plugin()->url(), $result->plugin()->path(), $script->src );
+
+			if ( ! in_array( $handle, wp_scripts()->in_footer, true ) ) {
+				$this->add_result_warning_for_file(
+					$result,
+					sprintf(
+						/* translators: 1: tested URL. 2: 'defer'. 3: 'async' */
+						__( 'This script on %1$s is potentially blocking. Consider a %2$s or %3$s script strategy or moving it to the footer.', 'plugin-check' ),
+						$url,
+						'defer',
+						'async'
+					),
+					'NonBlockingScripts.BlockingHeadScript',
+					$script_path
+				);
+			} else {
+				$this->add_result_warning_for_file(
+					$result,
+					sprintf(
+						/* translators: 1: tested URL. 2: 'defer'. 3: 'async' */
+						__( 'This script on %1$s is loaded in the footer. Consider a %2$s or %3$s script loading strategy instead.', 'plugin-check' ),
+						$url,
+						'defer',
+						'async'
+					),
+					'NonBlockingScripts.NoStrategy',
+					$script_path
+				);
+			}
```

Comment:
Wouldn't it be particularly helpful for the script handle to be indicated?

```suggestion
			if ( ! in_array( $handle, wp_scripts()->in_footer, true ) ) {
				$this->add_result_warning_for_file(
					$result,
					sprintf(
						/* translators: 1: tested URL. 2: the script handle. 3: 'defer'. 4: 'async' */
						__( 'This script on %1$s (with handle %2$s) is potentially blocking. Consider a %3$s or %4$s script strategy or moving it to the footer.', 'plugin-check' ),
						$url,
						$handle,
						'defer',
						'async'
					),
					'NonBlockingScripts.BlockingHeadScript',
					$script_path
				);
			} else {
				$this->add_result_warning_for_file(
					$result,
					sprintf(
						/* translators: 1: tested URL. 2: the script handle.3: 'defer'. 4: 'async' */
						__( 'This script on %1$s (with handle %2$s) is loaded in the footer. Consider a %3$s or %4$s script loading strategy instead.', 'plugin-check' ),
						$url,
						$handle,
						'defer',
						'async'
					),
					'NonBlockingScripts.NoStrategy',
					$script_path
				);
			}
```

Reply:
In the admin/CLI output the annotation will be under the script file name, isn't that enough context?

Reply:
Yeah, but since the script handles are string literals that are used for registering and enqueueing scripts, where the src may not always be immediately apparent, it seems having the handle too would be useful context.